src/state/slices/portfolioSlice/selectors.ts

```diff
@@ -769,27 +769,77 @@ export const selectPortfolioAccountIdsSortedFiat = createDeepEqualOutputSelector
+      (acc, [accountId, account]) => {
+        Object.values(account?.stakingDataByValidatorId ?? {}).forEach(stakingDataByAssetId => {
+          Object.values(stakingDataByAssetId).forEach(stakingData => {
+            const { delegations, redelegations, undelegations } = stakingData
+            const redelegationEntries = redelegations.flatMap(redelegation => redelegation.entries)
+            const combined = [...delegations, ...redelegationEntries, ...undelegations]
+            combined.forEach(({ assetId, amount }) => {
+              if (!acc[accountId]) acc[accountId] = {}
+              acc[accountId][assetId] = bnOrZero(acc[accountId][assetId]).plus(amount).toString()
+            })
+          })
+        })
+        return acc
+      },
+      {},
+    )
+  },
+)
+
+/**
+ * returns crypto human staking amount by assetId and accountId filter
+ */
+export const selectPortfolioStakingCryptoHumanBalanceByFilter = createSelector(
+  selectAssets,
+  selectPortfolioStakingCryptoBalances,
+  selectAssetIdParamFromFilterOptional,
+  selectAccountIdParamFromFilterOptional,
+  (assets, stakingBalances, assetIdFilter, accountIdFilter): string => {
+    return Object.entries(stakingBalances)
+      .filter(([accountId]) => (accountIdFilter ? accountId === accountIdFilter : true))
+      .reduce<BigNumber>((acc, [, account]) => {
+        Object.entries(account)
+          .filter(([assetId]) => (assetIdFilter ? assetId === assetIdFilter : true))
+          .forEach(([assetId, balance]) => {
+            acc = acc.plus(bnOrZero(fromBaseUnit(bnOrZero(balance), assets[assetId].precision)))
+          })
+
+        return acc
+      }, bn(0))
+      .toString()
+  },
+)
+
 /**
  * selects all accounts in PortfolioAccountBalancesById form, including all
  * delegation, undelegation, and redelegation balances, with base unit crypto balances
  */
 export const selectPortfolioAccountsCryptoBalancesIncludingStaking = createDeepEqualOutputSelector(
-  selectPortfolioAccounts,
   selectPortfolioAccountBalances,
-  (accounts, accountBalances): PortfolioAccountBalancesById => {
-    return Object.entries(accounts).reduce((acc, [accountId, account]) => {
-      Object.values(account?.stakingDataByValidatorId ?? {}).forEach(stakingDataByAssetId => {
-        Object.values(stakingDataByAssetId).forEach(stakingData => {
-          const { delegations, redelegations, undelegations } = stakingData
-          const redelegationEntries = redelegations.flatMap(redelegation => redelegation.entries)
-          const combined = [...delegations, ...redelegationEntries, ...undelegations]
-          combined.forEach(({ assetId, amount }) => {
-            acc[accountId][assetId] = bnOrZero(acc[accountId][assetId]).plus(amount).toString()
-          })
+  selectPortfolioStakingCryptoBalances,
+  (accountBalances, stakingBalances): PortfolioAccountBalancesById => {
+    return Object.entries(accountBalances).reduce<PortfolioAccountBalancesById>(
```

Comment:
i moved the guts of this logic up to `selectPortfolioStakingCryptoBalances`, now this selector just does a deep merge

Reply:
by deep merge i mean deep sum merge